internal/webhooks/cluster.go

```diff
@@ -246,6 +247,26 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 		)
 	}
 
+	// upgrade concurrency should be a numeric value.
+	if concurrency, ok := newCluster.Annotations[clusterv1.ClusterTopologyUpgradeConcurrency]; ok {
+		concurrencyAnnotationField := field.NewPath(fmt.Sprintf("annotations[%q]", clusterv1.ClusterTopologyUpgradeConcurrency))
+		concurrencyInt, err := strconv.Atoi(concurrency)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(
+				concurrencyAnnotationField,
+				concurrency,
+				errors.Wrap(err, "could not parse the value of the annotations").Error(),
+			))
+		}
+		if concurrencyInt < 1 {
```

Comment:
I think we also hit this case if err is != nil as AtoI returns `0, err` in this case. I think this is confusing to users. Let's move this check in the else branch